src/schema/schema-generator.ts

```diff
@@ -373,15 +373,25 @@ export abstract class SchemaGenerator {
       if (handler.resolverClassMetadata && handler.resolverClassMetadata.isAbstract) {
         return fields;
       }
+      const pubSubIterator = (payload: any, args: any, context: any, info: any) => {
+        let topics: string | string[];
+        if (typeof handler.topics === "function") {
+          const resolverTopicData: ResolverTopicData = { payload, args, context, info };
+          topics = handler.topics(resolverTopicData);
+        } else {
+          topics = handler.topics as string[];
```

Comment:
this assertion is not needed

![image](https://user-images.githubusercontent.com/10618781/44874111-6db01f00-ac9a-11e8-8ad4-218fb1dfd720.png)